autoscale/src/main/java/com/sequenceiq/periscope/service/EntitlementValidationService.java

```diff
@@ -48,4 +48,20 @@ public boolean stopStartAutoscalingEntitlementEnabled(String accountId, String c
         }
         return entitled;
     }
+
+    @Cacheable(cacheNames = "accountEntitlementCacheImpalaSchedule", key = "{#accountId,#cloudPlatform}")
+    public boolean impalaScheduleBasedScalingEntitlementEnabled(String accountId, String cloudPlatform) {
+        boolean entitled = autoscalingEntitlementEnabled(accountId, cloudPlatform);
+        if (!entitled) {
+            return false;
+        }
+        if (CloudPlatform.AWS.equalsIgnoreCase(cloudPlatform)) {
+            entitled = entitlementService.awsImpalaScheduleScalingEnabled(accountId);
+        } else if (CloudPlatform.AZURE.equalsIgnoreCase(cloudPlatform)) {
+            entitled = entitlementService.azureImpalaScheduleScalingEnabled(accountId);
+        } else if (CloudPlatform.GCP.equalsIgnoreCase(cloudPlatform)) {
+            entitled = entitlementService.gcpImpalaScheduleScalingEnabled(accountId);
+        }
+        return entitled;
+    }
```

Comment:
Instead of using if-else like this, We should have Map<CloudPlatform,List<Entitlement>> which is predefined. Based on CloudPlatform, find the list of entitlements. check if all of them are assigned and put the missing ones in error message. 

You can refer to https://github.com/hortonworks/cloudbreak/blob/67830bcca379fe1a4c4326a39ddf2edd10f22712/environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/FreeIpaConverter.java

Reply:
CHanged